aiida/orm/implementation/general/calculation/work.py

```diff
@@ -43,4 +44,18 @@ def has_failed(self):
         :return: True if the calculation has failed, False otherwise.
         :rtype: bool
         """
-        return self.get_attr(self.FAILED_KEY, False) is not False
\ No newline at end of file
+        return (
+            self.get_attr(self.FAILED_KEY, False) or
+            self.get_attr(self.ABORTED_KEY, False)
+        )
+
+    def kill(self):
+        """
+        Kill a WorkCalculation and all its children.
+        """
+        if not self.is_sealed:
+            self._set_attr(self.ABORTED_KEY, True)
+        self.seal()
+        for key, child in self.get_outputs_dict().items():
```

Comment:
You can pass the desired link type to `get_outputs_dict(link_type=LinkType.CALL)` which is safer than relying on the label